prober/derp.go

```diff
@@ -204,6 +213,24 @@ func (d *derpProber) probeMapFn(ctx context.Context) error {
 	return nil
 }
 
+func inSameVPC(a, b *tailcfg.DERPNode) bool {
+	return vpcName(a.Name) == vpcName(b.Name)
+}
+
+// vpcName returns the name of a VPC that a DERP Node is running
+// in as a function of its DERPNode.Name.
+//
+// This is a naming convention for the Tailscale-run DERP nodes only
+// and is not a thing that's supported by clients. See the comments
+// at the caller of this.
+func vpcName(n string) string {
+	_, name, ok := strings.Cut(n, "-vpc")
```

Comment:
Are we going to update `derpmap` node names accordingly? I don't see this `-vpc` part in existing derp names.
Also, WDYT about including a separate `VPC` (or `MeshGroup`) field in `tailcfg.DERPNode` instead of encoding that info in the name?